### PR TITLE
QuickSort 3-Way in PHP

### LINKED ISSUE #2170: Quick Sort 3 way
- [ ] C - @Moddy2024
- [x] C++
- [ ] C# - @Harikrishnan6336
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [ ] Java - @IPSITA2000
- [ ] JS - @atrip10
- [ ] Kotlin
- [ ] Php - @BellatrixLestrangee
- [x] Python
- [ ] Readme - @SomyaGandhi
- [x] Ruby - @HariniJeyaraman
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by BellatrixLestrangee**
Fixes #2170 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Adding 3-Way QuickSort Algorithm in PHP
- 

#### Languages Used:
- PHP
- 

#### Files Added:
- Quicksort_3_way.php
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by BellatrixLestrangee** [comment]
@asha15 I have changed the PR to cover a single issue as you asked in #2502. Please review the pull request.

File: Quicksort_3_way/Quicksort_3_way.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+{ 
+	$i = $l-1;
+	$j = $r; 
+	$p = $l-1;
+	$q = $r; 
+	$v = $a[$r]; 
+
+	while (true) 
+	{ 
+		// From left, find the first element greater than
```
Comment (singh-shreya6) [inline Quicksort_3_way/Quicksort_3_way.php]: 4 space indentation please.

**Comment by BellatrixLestrangee** [comment]
@singh-shreya6 Ma'am, I have done the changes. Please kindly check.

File: Quicksort_3_way/Quicksort_3_way.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+// A utility function to print an array 
+function printarr($a, $n) 
+{ 
+	for ($i = 0; $i < $n; ++$i) 
+		echo $a[$i]." "; 
+	echo "\xA"; 
+} 
+
+// Driver program 
+
```
Comment (singh-shreya6) [inline Quicksort_3_way/Quicksort_3_way.php:100]: where is the function line for this?

Comment (BellatrixLestrangee) [reply Quicksort_3_way/Quicksort_3_way.php:100]: In PHP, we don't require any explicit main function.

**Comment by BellatrixLestrangee** [comment]
This time i have manually checked and given spaces along each line, i hope it's alright. @singh-shreya6

File: Quicksort_3_way/Quicksort_3_way.php (hunk 2 above)
Comment (BellatrixLestrangee) [reply Quicksort_3_way/Quicksort_3_way.php:100]: Enclosed the driver program in brackets for clarity.

Comment (singh-shreya6) [reply Quicksort_3_way/Quicksort_3_way.php:100]: sure.

File: Quicksort_3_way/Quicksort_3_way.php [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    // Move all left same occurrences from beginning 
+    // to adjacent to arr[i] 
+    $j = $i-1; 
+    for ($k = $l; $k < $p; $k++, $j--) 
+        swap($a[$k], $a[$j]); 
+
+    // Move all right same occurrences from end 
+    // to adjacent to arr[i] 
+    $i = $i+1; 
+    for ($k = $r-1; $k > $q; $k--, $i++)
```
Comment (singh-shreya6) [inline Quicksort_3_way/Quicksort_3_way.php:73]: Add space around operators.

File: Quicksort_3_way/Quicksort_3_way.php [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    // Move all left same occurrences from beginning 
+    // to adjacent to arr[i] 
+    $j = $i-1; 
+    for ($k = $l; $k < $p; $k++, $j--) 
+        swap($a[$k], $a[$j]); 
+
+    // Move all right same occurrences from end 
+    // to adjacent to arr[i] 
+    $i = $i+1;
```
Comment (singh-shreya6) [inline Quicksort_3_way/Quicksort_3_way.php:72]: Around + here and around - in next line.

File: Quicksort_3_way/Quicksort_3_way.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+function swap(&$a, &$b)
+{
+    $temp = $a;
+    $a = $b ;
+    $b = $temp;
+}
+
+function partition(&$a, $l, $r, &$i, &$j) 
+{ 
+    $i = $l-1;
```
Comment (singh-shreya6) [inline Quicksort_3_way/Quicksort_3_way.php:18]: Space around -

File: Quicksort_3_way/Quicksort_3_way.php [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    $j = $r; 
+    $p = $l-1;
```
Comment (singh-shreya6) [inline Quicksort_3_way/Quicksort_3_way.php:20]: Space around -

**Comment by BellatrixLestrangee** [comment]
I am not sure how to squash, i'll be glad if you could help.

**Comment by BellatrixLestrangee** [comment]
I'll make a new fork and try again.

**Comment by BellatrixLestrangee** [comment]
close this pr

**Pull request closed without merging**